InfrastructureBackupValidator/BackupValidationTool.psm1

```diff
@@ -0,0 +1,591 @@
+            if ($plan2offer.ContainsKey(($planId)))
+            {
+                $plan2offer[$planId][$planLinkType] += $offerId
+            }
+            else
+            {
+                $plan2offer[$planId] = $emptyType2Offer
+                $plan2offer[$planId][$planLinkType] += $offerId
+            }
+        }
+
+        $SQLCmd = "SELECT [Id],[ResellerSubscriptionId],[ResourceGroupName],[Tags],[Name],
+            [DisplayName],[Description],[ProvisioningState],[RoutingResourceManagerType]
+            FROM [$SubscriptionSqlDbName].[subscriptions.internal].[Plans]"
+        $planTable = Invoke-Sqlcmd -Database $SubscriptionSqlDbName -Query $SQLCmd -As DataSet @sqlCommonParams
+        $plans = @{}
+        $planColumnNames = $planTable.Tables[0].Columns.ColumnName
+        foreach ($row in $planTable.Tables[0].Rows)
+        {
+            $planId = ""
+            $plan = [ordered] @{}
+            foreach ($column in $planColumnNames)
+            {
+                $value = $row[$column].ToString()
+                if ($column -eq "ResellerSubscriptionId")
+                {
+                    $plan[$column] = $resellerSubcriptions[$value]
+                }
+                elseif ($column -eq "ProvisioningState")
+                {
+                    $plan[$column] = $provisioningStates[$value]
+                }
+                elseif ($column -eq "RoutingResourceManagerType")
+                {
+                    $plan[$column] = $resourceManagerTypes[$value]
+                }
+                elseif ($column -eq "Id")
+                {
+                    $planId = $value
+                }
+                else
+                {
+                    $plan[$column] = $value
+                }
+            }
+
+            $plan["Id"] = "/subscriptions/$($plan["ResellerSubscriptionId"])/resourceGroups/$($plan["ResourceGroupName"])/providers/Microsoft.Subscriptions.Admin/plans/$($plan["Name"])"
+            $plan["QuotaIds"] = $plan2quota[$planId]
+            $plan["Offers"] = $plan2offer[$planId]
+            $plans.Add($planId, $plan)
+        }
+
+        Write-Verbose $Strings.MsgPlan -Verbose
+        foreach ($plan in $plans.Values)
+        {
+            Write-Host ($plan | Out-String)
+            Write-Host "`n"
+        }
+    }
+    finally
+    {
+        # Drop subscription DB        
+        $disconnectScript = @"
+DECLARE @DatabaseName nvarchar(50)
+SET @DatabaseName = N'$SubscriptionSqlDbName'
+DECLARE @SQL varchar(max)
+SELECT @SQL = COALESCE(@SQL,'') + 'Kill ' + Convert(varchar, SPId) + ';'
+FROM MASTER..SysProcesses
+WHERE DBId = DB_ID(@DatabaseName) AND SPId <> @@SPId
+--SELECT @SQL
+EXEC(@SQL)
+"@
+
+        Invoke-Sqlcmd -Query $disconnectScript @sqlCommonParams -ErrorAction Continue
+        Invoke-Sqlcmd -Query ('Drop database "{0}"' -f $SubscriptionSqlDbName) @sqlCommonParams -ErrorAction Continue
+    }
+
+    return @{
```

Comment:
The returned value is a print-out of the things from the backup, is that what P0 is about? i.e., is it up to the customers themselves to compare these values against their running envs?